robot/src/main/java/frc/robot/subsystems/Drivetrain.java

```diff
@@ -0,0 +1,188 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package frc.robot.subsystems;
+
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+
+import edu.wpi.first.wpilibj.command.Subsystem;
+import edu.wpi.first.wpilibj.drive.DifferentialDrive;
+import frc.robot.Constants;
+import frc.robot.commands.DrivetrainTeleop;
+
+/**
+ * Add your docs here.
+ */
+public class Drivetrain extends Subsystem {
+
+    // Generally, return a singleton instance of the subsystem
+    private static Drivetrain sInstance;
+
+    public static Drivetrain getInstance() {
+        if (sInstance == null) {
+            sInstance = new Drivetrain();
+        }
+
+        return sInstance;
+    }
+    
+    // Motor controller declarations
+    private CANSparkMax mLeftMaster, mLeftSlave, mRightMaster, mRightSlave;
```

Comment:
I'd rather go with `leader`/`follower` phrasing here. Feels like it fits better with the `.follow(...)` method.